ddl/cache.lua

```diff
@@ -0,0 +1,124 @@
+local cache = nil
+
+local SPACE_NAME_IDX      = 1
+local SHARD_FUNC_NAME_IDX = 2
+local SHARD_FUNC_BODY_IDX = 3
+
+-- Build cache.
+--
+-- We don't need to call this function with any type of locking:
+-- _ddl_sharding_func is memtx space, so calling :pairs() on it
+-- is atomic
+--
+-- Cache structure format:
+--
+-- cache = {
+--   spaces = {
+--     [space_name] = {
+--       -- raw sharding metadata, used for ddl.get()
+--       raw_tuple = <tuple object> (<nil> at error),
+--       -- parsed dot notation (like {'foo', 'bar'})
+--       parsed_func_name = <table> or <nil>
+--       -- a function ready to call
+--       callable = <function> or <nil>,
+--       -- string with an error: not nil only if setting callable fails
+--       error = <string> or <nil>,
+--     }
+--   },
+--   -- current schema version
+--   schema_version = <...>,
+-- }
```

Comment:
Nit: Two spaces instead of four used in indentation.

Reply:
fixed

Reply:
(Looks resolved, so I'm going to resolve this discussion.)